toptal/picasso
# feat(Drawer): add BackdropProps prop

[CC-839](https://toptal-core.atlassian.net/browse/CC-839)
Add possibility to pass props to backdrop overlay for Drawer

### Description

Add extra prop to the `Drawer` - `BackdropProps`. We can use it like `BackdropProps={{ invisible: true }}`. In this case, overlay will be transparent without dark background, but you can still click on the backdrop to hide `Drawer`.

### How to test

Open storybook for the Drawer and check the last example `With invisible backdrop`

### Screenshots

https://user-images.githubusercontent.com/16545305/236804196-cb9f2fcb-ffab-4718-ba1e-17eb70507d05.mov

### Development checks


**Breaking change**


> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[CC-839]: https://toptal-core.atlassian.net/browse/CC-839?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
